workspaces/ai-integrations/plugins/ai-experience/src/utils/rss-utils.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export const extractImageFromHTML = (html: string): string | undefined => {
+  if (!html) return undefined; // Explicitly return undefined
+  const match = html.match(/<img[^>]+src=(?:"([^">]+)"|'([^'>]+)')/i);
+  return match?.[1] || match?.[2];
+};
+
+export const sanitizeXML = (xml: string): string => {
+  return xml.replace(
+    /&(?!amp;|lt;|gt;|quot;|apos;|#\d+;|#[xX][0-9a-fA-F]+;)/g,
+    '&amp;',
+  );
```

Comment:
In Description single quote is not getting santized, can we include that?

![image](https://github.com/user-attachments/assets/09bceb2e-2588-467f-9969-cd64cd9ad37e)

you can use `https://techcrunch.com/feed` to reproduce this